libs/algolia/useFacets.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { useSearchContext } from './context';
+import { FacetItemsByValue } from './types';
+
+export type Facet = {
+   name: string;
+   itemsByValue: FacetItemsByValue;
+};
+
+export function useFacets(): Facet[] {
+   const { searchResult } = useSearchContext();
+
+   const facets = React.useMemo<Facet[]>(() => {
+      return Object.keys(searchResult.facetsByName).map((facetName) => {
+         return {
+            name: facetName,
+            itemsByValue: searchResult.facetsByName[facetName],
+         };
+      });
```

Comment:
Minor, but I think you could simplify this to:
```suggestion
      return searchResult.facetsByName.map((facet, facetName) => {
         return {
            name: facetName,
            itemsByValue: facet,
         };
      });
```